src/ucs/sys/string.c

```diff
@@ -484,3 +484,20 @@ ucs_status_t ucs_string_alloc_formatted_path(char **buffer_p, const char *name,
     *buffer_p = temp_buffer;
     return UCS_OK;
 }
+
+ucs_status_t ucs_string_alloc_path_buffer_and_get_dirname(char **buffer_p,
+                                                          const char *name,
+                                                          const char *path,
+                                                          char **dir_p)
```

Comment:
maybe it could be const char**, to avoid changes in the functions that declare dir name variable as "const char *"?
also it signifies that we do not allow changing/releasing *dir_p

Reply:
done